lnwallet/channel.go

```diff
@@ -1958,6 +1965,12 @@ func (lc *LightningChannel) restoreStateLogs(
 	for i := range remoteCommitment.outgoingHTLCs {
 		htlc := remoteCommitment.outgoingHTLCs[i]
 
+		if pendingRemoteCommit != nil {
```

Comment:
nit: prob don't need the nil check

Reply:
need the nil check as otherwise the code checks for `htlc.LogIndex` in `pendingRemoteOutgoingIndices` and won't find it, and will skip adding the htlc to update logs

Reply:
OK, should add why we skip it to a comment.

Reply:
Done